### PR TITLE
Generic fix

### PR DESCRIPTION
**Pull request by TomyLobo**
Fixed some warnings:
- Removed some redundant null checks and casts.
- Removed some redundant superinterfaces.
- Removed some unnecessary "throws" declarations.
- Removed some unnecessary casts.

Some other changes:
- Split up SignBase.canSupport to 2 subclasses: WallSign and SignPost.
- Removed the @Ignore on BaseProtocolTest and made it abstract instead.
- Made some control flow easier to follow, mostly by extracting else bodies and flipping if/else conditions

### CODE REVIEW

**Comment by Zidane** [comment]
Your changes look alright (I would attempt to squash them down some as pulling 7 commits at once scares me a bit...) but I am concerned by two files In this bunch without headers. This tells me that you either...

A. Provided custom maven flags (mvn clean install for example).
B. You never compiled this (and consequently never tested it).

File: src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java [hunk 1]
```diff
@@ -597,12 +597,12 @@ public boolean stopDigging(Point position) {
...
-		if (isDigging) {
-			// Is this correct?
-			return System.currentTimeMillis() - diggingStartTime;
-		} else {
+		if (!isDigging) {
 			return previousDiggingTime;
 		}
+
+		// Is this correct?
+		return System.currentTimeMillis() - diggingStartTime;
```
Comment (LunNova) [inline src/main/java/org/spout/vanilla/controller/living/player/VanillaPlayer.java:680]: What?!

edit: Never mind, I can't read diffs. :P

**Comment by TomyLobo** [comment]
@NinjaZidane I have no idea how you conclude missing testing from missing license blocks

but you're right :)
I just put this here for reference and so people can start taking a look if they want to
I didn't sign them off anyway. sorry should have mentioned it, but... too tired

**Comment by TomyLobo** [comment]
I ran the unit tests (which added the license blocks)
I see how you made that conclusion now :P
no manual testing done yet

**Comment by LunNova** [comment]
Because the maven clean stage adds licenses - even just compiling should result in the license being added.

**Comment by TomyLobo** [comment]
I'm usually compiling with eclipse :)

**Comment by LunNova** [comment]
Use Eclipse's maven integration and make it use the correct goals.

**Comment by TomyLobo** [comment]
what else do you think I'm using -_-
it's just that eclipse doesn't invoke any maven goals when I press Ctrl-B

**Comment by TomyLobo** [comment]
here's the result of my manual testing:
- got an exception on startup:
  13:23:47 [SCHWERWIEGEND] Unable to load 'WorldEdit.jar' in directory 'plugins': The field 'platform' is not present in the plugin.yml!
  org.spout.api.exception.InvalidDescriptionFileException: The field 'platform' is not present in the plugin.yml!
      at org.spout.api.plugin.PluginDescriptionFile.getEntry(PluginDescriptionFile.java:169)
      at org.spout.api.plugin.PluginDescriptionFile.load(PluginDescriptionFile.java:113)
      at org.spout.api.plugin.PluginDescriptionFile.<init>(PluginDescriptionFile.java:82)
      at org.spout.api.plugin.CommonPluginLoader.getDescription(CommonPluginLoader.java:237)
      at org.spout.api.plugin.CommonPluginLoader.loadPlugin(CommonPluginLoader.java:132)
      at org.spout.api.plugin.CommonPluginManager.loadPlugin(CommonPluginManager.java:125)
      at org.spout.api.plugin.CommonPluginManager.loadPlugins(CommonPluginManager.java:175)
      at org.spout.engine.SpoutEngine.loadPlugins(SpoutEngine.java:243)
      at org.spout.engine.SpoutEngine.start(SpoutEngine.java:211)
      at org.spout.engine.SpoutServer.start(SpoutServer.java:95)
      at org.spout.engine.SpoutServer.main(SpoutServer.java:90)
- mobs arent moving correctly
- i think the spout server sent me half-generated columns and then later the full chunks
- <s>there are no ores</s>
- something that looks like a swamp biome (dark grass and such) doesn't generate. it's cut off to sea level with a half-generated lake in it. the rest didnt generate at all, but nothing else generated either. I reconnected to fix the issue but that got me back to the spawn :/ at least i got new chunks though :)
- oceans dont look very oceanlike either
- got an out-of-memory error:
  13:43:38 [SCHWERWIEGEND] Error while pulsing thread SpoutRegion{ ( 1, 0, 0), World: SpoutWorld{ world UUID: a9296f1e-8fa4-42d3-8bad-aeeecb2b3f37 Age: 864789}} Thread:  Java heap space
  java.lang.OutOfMemoryError: Java heap space
      at java.util.concurrent.atomic.AtomicIntegerArray.<init>(Unknown Source)
      at org.spout.api.util.map.concurrent.AtomicShortArray.<init>(AtomicShortArray.java:47)
      at org.spout.api.util.map.concurrent.AtomicBlockStoreImpl.<init>(AtomicBlockStoreImpl.java:77)
      at org.spout.engine.world.SpoutChunk.<init>(SpoutChunk.java:190)
      at org.spout.engine.world.FilteredChunk.<init>(FilteredChunk.java:62)
      at org.spout.engine.world.FilteredChunk.<init>(FilteredChunk.java:58)
      at org.spout.engine.world.SpoutRegion.generateChunk(SpoutRegion.java:322)
      at org.spout.engine.world.SpoutRegion.getChunk(SpoutRegion.java:260)
      at org.spout.engine.world.SpoutWorld.getChunk(SpoutWorld.java:321)
      at org.spout.engine.world.SpoutWorld.getChunkFromBlock(SpoutWorld.java:333)
      at org.spout.engine.world.SpoutWorld.getChunkFromBlock(SpoutWorld.java:343)
      at org.spout.engine.world.SpoutWorld.getChunkFromBlock(SpoutWorld.java:338)
      at org.spout.engine.world.SpoutWorld.getChunkFromBlock(SpoutWorld.java:97)
      at org.spout.api.protocol.NetworkSynchronizer.preSnapshot(NetworkSynchronizer.java:256)
      at org.spout.vanilla.protocol.VanillaNetworkSynchronizer.preSnapshot(VanillaNetworkSynchronizer.java:329)
      at org.spout.engine.entity.EntityManager.preSnapshotRun(EntityManager.java:231)
      at org.spout.engine.world.SpoutRegion.preSnapshotRun(SpoutRegion.java:801)
      at org.spout.engine.world.SpoutRegionManager.preSnapshotRun(SpoutRegionManager.java:71)
      at org.spout.engine.util.thread.coretasks.PreSnapshotTask.call(PreSnapshotTask.java:40)
      at org.spout.engine.util.thread.ThreadAsyncExecutor.executeTask(ThreadAsyncExecutor.java:168)
      at org.spout.engine.util.thread.ThreadAsyncExecutor.executeAllTasks(ThreadAsyncExecutor.java:163)
      at org.spout.engine.util.thread.ThreadAsyncExecutor.pulsedRun(ThreadAsyncExecutor.java:243)
      at org.spout.engine.util.thread.PulsableThread.run(PulsableThread.java:131)

**Comment by Zidane** [comment]
@TomyLobo

Making a comment like that here helps no one. Please make relevant JIRA issues.

To summarize...
1. WorldEdit has no Platform: SERVER, Platform: CLIENT, Platform: BOTH in its properties.yml
2. There are no ores - Decorators are off.
3. Mobs don't move correctly - Collisions and gravity/etc AI (Actions) are off.
4. Talk to @DDoSQc about improvements to WGEN.

**Comment by TomyLobo** [comment]
@NinjaZidane: @Olloth asked me to do some manual testing with my patches before he pulls them. these are the results.

**Comment by TomyLobo** [comment]
rebased to latest master and signed off all commits.

**Pull request merged**